app-testsuite/app/src/App.js

```diff
@@ -1,36 +1,54 @@
 import '@binaris/shift-code-transform/macro';
-import React, { useState, useEffect } from 'react';
-import { get, update } from '../backend/index';
+import React, { useState, useEffect, useCallback } from 'react';
+import { get, update, getSecret } from '../backend/index';
 
-const Counter = ({ keyName }) => {
+const Stateful = ({ getter, children }) => {
   const [{ loading, data, error }, setState] = useState({ loading: true });
   useEffect(() => {
     if (loading) {
-      get(keyName)
-        .then(count => setState({ data: count || 0 }))
+      getter()
```

Comment:
This would be easier to read as an `async` function, I think.  Especially if ever we wanted to add something to one of the portions.

Reply:
Agree, but I'd rather not delay this PR.